bigbluebutton-html5/imports/ui/components/whiteboard/utils.js

```diff
@@ -212,11 +212,90 @@ const mapLanguage = (language) => {
   }
 };
 
+const getFontStyle = (style) => {
+  const fontSizes = {
+    small: 28,
+    medium: 48,
+    large: 96,
+    auto: 'auto',
+  };
+
+  const fontFaces = {
+    script: '"Caveat Brush"',
+    sans: '"Source Sans Pro"',
+    serif: '"Crimson Pro"',
+    mono: '"Source Code Pro"',
+  }
+
+  const fontSize = fontSizes[style.size];
+  const fontFace = fontFaces[style.font];
+  const { scale = 1 } = style;
+
+  return `${fontSize * scale}px/1 ${fontFace}`;
+}
+
+const getMeasurementDiv = (font) => {
+  // A div used for measurement
+  document.getElementById('__textMeasure')?.remove();
+
+  const pre = document.createElement('pre');
+  pre.id = '__textMeasure';
+
+  Object.assign(pre.style, {
+    whiteSpace: 'pre',
+    width: 'auto',
+    border: '1px solid transparent',
+    padding: '4px',
+    margin: '0px',
+    letterSpacing: '-0.03em',
+    opacity: '0',
+    position: 'absolute',
+    top: '-500px',
+    left: '0px',
+    zIndex: '9999',
+    pointerEvents: 'none',
+    userSelect: 'none',
+    alignmentBaseline: 'mathematical',
+    dominantBaseline: 'mathematical',
+    font,
+  });
+
+  pre.tabIndex = -1;
+
+  document.body.appendChild(pre);
+  return pre;
+}
+
+const getTextSize = (text, style, padding) => {
+  const font = getFontStyle(style);
+
+  if (!text) {
+    return [16, 32];
+  }
+
+  const melm = getMeasurementDiv(font);
```

Comment:
I'd normally recommend creating the text measurement div only once at app startup, rather than every time - but showing poll results happens so infrequently that this is not really a performance concern. I'm OK with it.